CMakeLists.txt

```diff
@@ -23,16 +33,26 @@ if(CMAKE_SYSTEM_PROCESSOR)
 		set(ARM ON)
 		if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^armv7")
 			set(ARMV7 ON)
+			# Horrifying workaround for bug in android cmake stuff for asm files
+			if (ANDROID)
+				set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -target armv7a-none-linux-android")
+			endif()
 		endif()
-	elseif(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^x86" OR
-		${CMAKE_SYSTEM_PROCESSOR} MATCHES "^amd64" OR
-		${CMAKE_SYSTEM_PROCESSOR} MATCHES "i.86")
+	elseif(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^aarch64")
+	  set(ARM64 ON)
+	  add_definitions(-DARM64)
+	elseif(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^amd64" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "^x86_64")
+    set(X86_64 ON)
```

Comment:
nit: A new indentation crime in this indentation war-stricken file...

Maybe we need a `[CMakeLists.txt]` rule in editorconfig?  There's vim support:
https://github.com/editorconfig/editorconfig-vim

-[Unknown]